File: 加密保护/The Enigma Protector v2.70 Build 20110629/Examples/Trial/Delphi/test_unit.pas
unit test_unit;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, ExtCtrls;

type
  TForm1 = class(TForm)
    Label1: TLabel;
    Label2: TLabel;
    btnClose: TButton;
    Panel1: TPanel;
    Label3: TLabel;
    Label4: TLabel;
    Label5: TLabel;
    Label6: TLabel;
    Label7: TLabel;
    trDaysAll: TLabel;
    trDaysLeft: TLabel;
    trExecAll: TLabel;
    trExecLeft: TLabel;
    trDate: TLabel;
    procedure btnCloseClick(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

uses enigma_ide;

{$R *.dfm}

procedure TForm1.btnCloseClick(Sender: TObject);
begin
  Application.Terminate;
end;

procedure TForm1.FormShow(Sender: TObject);
var
  Total, Left : DWord;
  Day, Month, Year : DWord;
begin
  if EP_TrialDays(Total, Left) then
  begin
    trDaysAll.Caption := IntToStr(Total);
    trDaysLeft.Caption := IntToStr(Left);
  end;
  if EP_TrialExecutions(Total, Left) then
  begin
    trExecAll.Caption := IntToStr(Total);
    trExecLeft.Caption := IntToStr(Left);
  end;
  if EP_TrialExpirationDate(Year, Month, Day) then
  begin
    trDate.Caption := Format('%d-%d-%d', [Day, Month, Year]);
  end;
end;

end.
 

File: 加密保护/The Enigma Protector v2.33 Build 20110112/Examples/Trial/Bcb/test_unit.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#include <stdio.h>
#pragma hdrstop

#include "test_unit.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"

// Enigma API respondence ---------------------------------------------------
#include "../../../EnigmaSDK/Bcb/enigma_ide.h"
#pragma link "../../../EnigmaSDK/Bcb/enigma_ide.lib"
// --------------------------------------------------------------------------

TfrmTrial *frmTrial;
//---------------------------------------------------------------------------
__fastcall TfrmTrial::TfrmTrial(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmTrial::FormShow(TObject *Sender)
{
  DWORD Total = 0;
  DWORD Left = 0;
  WORD Day = 0;
  WORD Month = 0;
  WORD Year = 0;
  char buffer[255];

  if (EP_TrialDays(&Total, &Left))
  {
    lbDays->Caption = IntToStr(Total);
    lbDaysLeft->Caption = IntToStr(Left);
  }
  if (EP_TrialExecutions(&Total, &Left))
  {
    lbExecs->Caption = IntToStr(Total);
    lbExecsLeft->Caption = IntToStr(Left);
  }

  if (EP_TrialExpirationDate(&Year, &Month, &Day))
  {
    lbDate->Caption = Format("%d-%d-%d", ARRAYOFCONST((Day, Month, Year)));
  }
}


void __fastcall TfrmTrial::btnCloseClick(TObject *Sender)
{
  Close();
}
//---------------------------------------------------------------------------

